app/src/main/java/com/fastlink/zekrapp/presentation/ZekirScreen.kt

```diff
@@ -0,0 +1,211 @@
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.FabPosition
+import androidx.compose.material.FloatingActionButton
+import androidx.compose.material.FloatingActionButtonDefaults
+import androidx.compose.material.Scaffold
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowForward
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.platform.LocalClipboardManager
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.fastlink.zekrapp.LocalNavController
+import com.fastlink.zekrapp.LocalViewModel
+import com.fastlink.zekrapp.R
+import com.fastlink.zekrapp.presentation.utils.DashedDevider
+import com.fastlink.zekrapp.presentation.utils.AppBar
+import com.fastlink.zekrapp.presentation.utils.archBorder
+import com.fastlink.zekrapp.presentation.utils.bottomAppBar.BottomBar
+import com.fastlink.zekrapp.presentation.utils.bottomAppBar.getListOfBottomBarItemsInZekirScreen
+import com.fastlink.zekrapp.presentation.utils.dashedBorder
+import com.fastlink.zekrapp.presentation.utils.halfCircle
+
+
+@SuppressLint("RememberReturnType")
+@Composable
+fun ZekirScreen(categoryId: Int) {
+    val viewModel = LocalViewModel.current
+    val navController = LocalNavController.current
+    val clipboardManager = LocalClipboardManager.current
+    val scope = rememberCoroutineScope()
+    remember { viewModel.getZekirByCategoryId(categoryId) }
+    val category = viewModel.getCategoryById(categoryId)
+    val zekirNumber = viewModel.zekirNumber
+    val zekirCounter = viewModel.zekirCounter
+    val animatedBorder by animateFloatAsState(
+        targetValue =
+        (zekirCounter.intValue.toFloat() / viewModel.zekirs[zekirNumber.intValue].numericCounter.toFloat()) * 360f,
+        label = "",
+        animationSpec = tween(1000)
+    )
+
+    fun buttonIsEnable(): Boolean {
```

Comment:
No point for this function to be here, you can place it in your view mode.

Reply:
@bilalSerwan This hasn't changed yet. Please request review only when you're finished.